### DISCUSSION TITLE
Cannot delete Bucket after all object is deleted with golang minio client

### DISCUSSION THREAD

**Discussion by szabokacsa** [minio/minio#14134] [Q&A] [answered]
Hi,
I tried to remove all object from the Bucket then I wanted to remove the bucket, but it failed with error: 
"The bucket you tried to delete is not empty. You must delete all versions in the bucket."

The bucket was created with enabled objectlocking, because later the bucket will use replication.

This is the goland code what lists and removes  all objects from the bucket:

![image](https://user-images.githubusercontent.com/3647735/150181430-830527d3-e5ec-42dc-b710-76742205b9e8.png)

This is the goland code what removes the bucket (I tried to use force bucket removal before, but is not possible, because objectlocking is enabled):

![image](https://user-images.githubusercontent.com/3647735/150181322-f93b7a87-1631-43de-b67d-dc2777f58054.png)

I listed the content of the minio bucket before I run the object removal:
![image](https://user-images.githubusercontent.com/3647735/150180017-ee77f830-dbd7-4978-84d5-976ce8d79fe8.png)

I listed the content of the minio bucket after I run the object removal:
![image](https://user-images.githubusercontent.com/3647735/150180100-623badee-03e7-4e5c-af6d-dfa125b267f5.png)

This is the log about the run:

![image](https://user-images.githubusercontent.com/3647735/150181920-43a547fe-c0aa-4790-9c0b-12e3d6983557.png)

Then I run the object removal one more time:
![image](https://user-images.githubusercontent.com/3647735/150180191-9ed16287-7589-4295-b26b-4b0b3541bb23.png)

I'm not sure that I'm using the client properly.

Does somebody have some idea how to remove all object and delete bucket? Or how to force bucket removal even if the bucket is used with replication?

**Comment by klauspost** [comment]
@szabokacsa 

Maybe the objects are locked. Try `minio.RemoveObjectsOptions{GovernanceBypass: true}`

**Reply by szabokacsa** [reply]
Same result.

**Comment by klauspost** [comment, marked as answer]
Ah. You are using the unversioned delete.

You must send the version id: `client.RemoveObject(...., minio.RemoveObjectOptions{VersionID: object.VersionID})`

(or use ` client.RemoveObjects` - you can see an example here: https://github.com/minio/warp/blob/b8830bc9cdf30ea616106da117696b4ead51592f/pkg/bench/benchmark.go#L163)
